### ISSUE TITLE
HttpData source and HttpProxy destination

### ISSUE THREAD

**Issue by dilipdhankecha2530** [eclipse-dataspaceconnector/DataSpaceConnector#1930]
# Bug Report

## Describe the Bug
We want to pass data from consumer to provider the case is consumer create an asset with policy and contract details and during asset creation the type of data address is HTTPDATA. Now provider want to send some data to consumer and doing the contarct negotiation process and initiate the transfer process with data address type HTTPPROXY. 
During the transfer process initiate we face the below error.
```
SEVERE 2022-09-09T11:51:48.068826591 Fatal error processing transfer request: 73fb10b0-bf0a-4976-8204-4a49082a535b. Error details: Failed to delegate data transfer to Data Plane: Cannot find a transfer Service that can handle HttpData source and HttpProxy destination

```
### Expected Behavior
Data must be pass to Consumer to provider.

**Comment by juliapampus** [comment]
@dilipdhankecha2530 Could you please provide a complete bug report (referring to the template) covering the used version, used extensions, executed API calls etc.?

**Comment by dilipdhankecha2530** [comment]
@juliapampus 
So we want to pass data from the consumer side to the provider side but when we use the **HttpData** at that time transfer has been initiated from the provider side to the consumer side (First hit the rest endpoint from the asset data address and once we got the response hit the destination data address API).

But now the use case is the consumer wants to pass the data to the provider so here I am confused about how to do this part with **HttpData**? 
or 
Do we need to use another data address type for this type of senarios?

**Comment by bscholtes1A** [comment]
Hello, 
the problem description is very confusing.
 
First, the consumer does not pass data to the provider. It is the other way around. 

Also this is the provider who creates the Asset, not the consumer. 

If you want to use the Data Plane as a http proxy to query the data, which is typically what you want to do when the provider exposes its data through a REST API, then the consumer will have to set the data type as `HttpProxy` in the DataRequest. 

This type of data exchange is covered in this integration test: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/cebe2cf3295796255d81239247009ca7706ec1c2/system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/dataspaceconnector/test/e2e/AbstractEndToEndTransfer.java#L48

**Comment by dilipdhankecha2530** [comment]
@bscholtes1A 
Yes, we are using ```HttpProxy``` for consumer DataRequest and use ```HttpData``` for the provider DataRequest while creating the asset.
Now How the flow would be in EDC that not able to understand could you please explain or give the idea behind this?

**Comment by jimmarino** [comment]
I am going to convert this to a discussion as it is not an issue. 

>> Now How the flow would be in EDC that not able to understand could you please explain or give the idea behind this?

Please ask more specific questions so that we can try and answer them. General "how-do-things-work" questions can usually be answered by looking at the documentation, samples, or tests. In your case, set a breakpoint on where the error is happening and perform some investigation so you can provide us with more detail and make it easier for us to help you.
